scripts/run_e2e_tests.py

```diff
@@ -170,6 +170,32 @@ def is_oppia_server_already_running():
             return True
     return False
 
+def run_ng_compilation():
+    """Runs webpack compilation.
+
+    Args:
+        source_maps: bool. Whether to compile with source maps.
+    """
+    max_tries = 2
+    ng_bundles_dir_name = 'dist/oppia-angular'
+
+    for _ in range(max_tries):
+        try:
+            managed_ng_compiler = (
+                servers.managed_ng_compiler())
+            with managed_ng_compiler as proc:
+                proc.wait()
+        except subprocess.CalledProcessError as error:
+            print(error.output)
+            sys.exit(error.returncode)
+            return
+        if os.path.isdir(ng_bundles_dir_name):
+            break
+    else:
+        # We didn't break out of the loop, meaning all attempts have failed.
+        print('Failed to complete ng compilation, exiting...')
+        sys.exit(1)
```

Comment:
Can this be introduced as a common function?

Reply:
Done. Now in common.py